src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -369,26 +394,69 @@ private static bool TryGetClassDeclarationList(INamedTypeSymbol typeSymbol, [Not
 
                     switch (propertyNameNode.Name.Identifier.ValueText)
                     {
-                        case "DefaultIgnoreCondition":
-                            options.DefaultIgnoreCondition = (JsonIgnoreCondition)Enum.Parse(typeof(JsonIgnoreCondition), propertyValueStr);
+                        case nameof(JsonSourceGenerationOptionsAttribute.DefaultIgnoreCondition):
+                            {
```

Comment:
Why change to "ignore" semantics here instead of "throw"? e.g. TryParse() vs. Parse().

Reply:
It adds confusion to the users when their code doesn't compile, and the source generator starts adding more errors to their error list.

At least that was my experience in https://github.com/dotnet/runtime/issues/52277

Reply:
Yes, by ignoring - we are leaving the error notification to Roslyn's compilation infrastructure.